### PR TITLE
[IMPROVE] : Replaced a tag with Links

### LINKED ISSUE #116: [Improve] : Replace the 'a' tags
Using 'a' tags directly affect the performance, since it's supposed to be a Single Page Application, the page should not reload in certain routings.

- [x] Replace A tags
- [x] Add custom loading feedback

### PR DESCRIPTION
**Pull request by Aashray446**
* Closes #116
* replace a tags with Link at menuBar
* Add lazyloading at the EventHeader section
* Improve padding and margin for EventHeader
* Replace inStyle background image with custom logic resulting in performance improvement
* Add image_cover class in EventModuleCss

### CODE REVIEW

**Comment by Aashray446** [comment]
Preview
PR-10

**With the acceptance of this PR, We need to later add a custom loader to give feedback to the user when some time taking item is loading**

**Comment by Aashray446** [comment]
@Dnouv Does this look good? It will improve the performance, suggested by the documentation of Next and React.

**Comment by Dnouv** [comment]
Thank you for the innovative fix @Aashray446 

This is definitely good to have it; however, as of now, this is very confusing when we click on a link and nothing shows up for a few seconds, it may lead to not good UX. I would suggest adding the custom loading as well while you are at this issue.

**Comment by Aashray446** [comment]
Sure, I will add that too. Glad you like the fix

Thanks

**Comment by Aashray446** [comment]
pr-11

@Dnouv  Do you like this loader ?

**Comment by Dnouv** [comment]
This looks amazing, nice UI work @Aashray446 

But I think UX-wise, there is room for improvement.

It is generally better to use a small loader in the corner and allow the user to interact with the old page until the new one appears. This is because users can become impatient if they are forced to wait for a page to load without any feedback and may perceive the website as slow or unresponsive.

By providing a small loader, you are indicating to the user that the page is loading and that they should wait for a moment. This can help manage user expectations and reduce frustration. Additionally, allowing the user to continue interacting with the old page can help minimize interruptions in their workflow.

Thank you! I think NextJS also shows a small Vercel logo in the bottom-right corner.

**Comment by Aashray446** [comment]
@Dnouv Thanks for the feedback, I will be back with some designs like that soon.

**Comment by Aashray446** [comment]
@Dnouv 
# Bottom Loader
design-1

# Top Loader
design-2

How does it look? Do you like anyone of them? Or do you want a loader at the corner same as the next js one?
The above type of loader will not interfere with the content of the page.

**Comment by Aashray446** [comment]
Hey @Dnouv, If it looks good, can you merge it?

File: app/pages/_app.js [hunk 1]
```diff
@@ -5,15 +5,23 @@ import SSRProvider from 'react-bootstrap/SSRProvider';
...
 import client from '../apollo-client';
+import { useState } from 'react';
+import { Router } from 'next/router';
 
 function MyApp({ Component, pageProps: {session, ...pageProps}}) {
+  const [loading, setLoading] = useState(false);
+
+  Router.events.on('routeChangeStart', () => setLoading(true))
+  Router.events.on('routeChangeComplete', () => setLoading(false))
+
```
Comment (Dnouv) [inline app/pages/_app.js:16]: Could we not do this inside the `<Layout />`?

**Review by Dnouv — CHANGES_REQUESTED** [review]
Please have a look if we could avoid making changes to the `_app.js`. Thank you!

**Comment by Aashray446** [comment]
Yeah sure ... I will do that , it's possible ig 
Thank you

File: app/pages/_app.js (hunk 1 above)
Comment (Aashray446) [reply app/pages/_app.js]: Thanks , I have applied the changes

**Comment by Aashray446** [comment]
@Dnouv, I have made the changes as you instructed. You can merge it now , I have rebased my feature branch.

Thanks

**Review by Dnouv — APPROVED** [review]
Thank you for the PR. LGTM!

**Pull request merged**